s3.go

```diff
@@ -29,8 +30,27 @@ import (
 	"github.com/minio/minio-go/v7/pkg/credentials"
 )
 
+// error messages
+const (
+	unableToInitializeConnection = "Unable to initialize connection to S3"
+	minioClientIsNil             = "Minio Client is nil"
+	wrongMinioClientReference    = "Wrong Minio client reference"
+	wrongBucketName              = "Wrong bucket name"
+	objectNameIsNotSet           = "Object name is not set"
+	wrongObjectName              = "Wrong object name"
+	bucketNameIsNotSet           = "Bucket name is not set"
+)
+
 // NewS3Connection function initializes connection to S3/Minio storage.
 func NewS3Connection(configuration *ConfigStruct) (*minio.Client, context.Context, error) {
+	// check if configuration structure has been provided
+	if configuration == nil {
+		err := errors.New("Configuration is nil")
+		log.Error().Err(err).Msg("Configuration error")
```

Comment:
Maybe you can set "Configuration is nil" and "Configuration error" as constants too?

Reply:
yup, nice catch, fixed